python/lsst/meas/extensions/shapeHSM/_hsm_moments.py

```diff
@@ -91,12 +108,11 @@ def _calculate(
         self,
         record: afwTable.SourceRecord,
         *,
-        image: galsim.Image,
-        weight: galsim.Image | None = None,
-        badpix: galsim.Image | None = None,
+        image: galsim._galsim.ImageViewD | galsim._galsim.ImageViewF,
+        weight_image: galsim._galsim.ImageViewI,
+        centroid: Point2D,
         sigma: float = 5.0,
         precision: float = 1.0e-6,
-        centroid: Point2D | None = None,
```

Comment:
Good catch. This routine doesn't really support `centroid` being None.

Reply:
It didn't before, and it still doesn't.  (It could be added, but I don't think that's necessary).